python/chemiscope/input.py

```diff
@@ -702,31 +702,41 @@ def _linearize(name, property, n_structures, n_centers, multidimensional=False):
     _validate_property(name, property)
 
     data = {}
+    values = property["values"]
     if not multidimensional:
-        if isinstance(property["values"], list):
-            data[name] = {
-                "target": property["target"],
-                "values": _typetransform(property["values"], name),
-            }
-        elif isinstance(property["values"], np.ndarray):
-            if len(property["values"].shape) == 1:
+        if isinstance(values, list):
+            if isinstance(values[0], list):
+                for i in range(len(values[0])):
+                    try:
+                        values_i = [v[i] for v in values]
+                    except IndexError:
+                        raise IndexError("List property has inconsistent size")
+                    data[f"{name}[{i + 1}]"] = {
+                        "target": property["target"],
+                        "values": _typetransform(values_i, name),
+                    }
```

Comment:
Why not use `np.array(values)` here, letting numpy do all the types & size checks for us?

Reply:
ok so you suggest to check if it's a list of lists, then convert to ndarray and go through the ndarray path? seemed hacky but you're right it simplifies the code a lot. will try this

Reply:
Have a look I've streamlined this